Firebase/Auth/FirebaseAuth.podspec

```diff
@@ -20,37 +20,60 @@ Simplify your iOS development, grow your user base, and monetize more effectivel
   s.source           = { :git => 'https://github.com/firebase/firebase-ios-sdk.git', :tag => s.version.to_s }
   s.social_media_url = 'https://twitter.com/Firebase'
   s.ios.deployment_target = '7.0'
+  s.osx.deployment_target = '10.10'
+
+  eitherSource = lambda { |paths|
+    Array(paths).map { |path| ['Firebase/Auth/Source/' + path, 'Source/' + path] }.flatten
+  }
+
+  s.source_files = eitherSource[[
+    'FirebaseAuth.h',
+    'FirebaseAuthVersion.h',
+    'FIRAdditionalUserInfo.h',
+    'FIRAuth.h',
+    'FIRAuthAPNSTokenType.h',
+    'FIRAuthCredential.h',
+    'FIRAuthDataResult.h',
+    'FIRAuthErrors.h',
+    'FIRAuthSwiftNameSupport.h',
+    'AuthProviders/EmailPassword/FIREmailAuthProvider.h',
+    'AuthProviders/Facebook/FIRFacebookAuthProvider.h',
+    'AuthProviders/GitHub/FIRGitHubAuthProvider.h',
+    'AuthProviders/Google/FIRGoogleAuthProvider.h',
+    'AuthProviders/OAuth/FIROAuthProvider.h',
+    'AuthProviders/Phone/FIRPhoneAuthCredential.h',
+    'AuthProviders/Phone/FIRPhoneAuthProvider.h',
+    'AuthProviders/Twitter/FIRTwitterAuthProvider.h',
+    'FIRUser.h',
+    'FIRUserInfo.h'
+  ]]
+
+  # Necessary hack to appease header visibility while as a direct OR transitive/internal dependency
+  s.subspec 'Internal' do |ss|
+    ss.source_files = eitherSource['**/*.[mh]']
+    ss.private_header_files = eitherSource['**/*.h']
+
+    ss.osx.exclude_files = eitherSource[[
+      'FIRAuthAppDelegateProxy.[mh]',
+      'FIRAuthNotificationManager.[mh]',
+      'FIRAuthAPNSTokenManager.[mh]',
+      'FIRAuthAPNSTokenType.[mh]',
+      'FIRAuthAPNSToken.[mh]',
+      'AuthProviders/Phone/FIRPhoneAuthProvider.[mh]'
+    ]]
+  end
 
-  s.source_files = '**/*.[mh]'
-  s.public_header_files =
```

Comment:
Removing the public headers specifires  is a problem. I'm seeing all the headers published in the framework instead of only the public ones, when I build with https://github.com/mhuusko5/firebase-ios-sdk/blob/master/BuildFrameworks/build.swift